packages/react/src/hooks/useAuth.ts

```diff
@@ -28,7 +28,7 @@ export const useAuth = (): UseAuthResult => {
    * This util will be used to get current user after those events.
    */
   const fetchCurrentUser = React.useCallback(async () => {
-    setResult({ user: undefined, isLoading: true, error: undefined });
+    setResult((result) => ({ ...result, isLoading: true }));
```

Comment:
To prevent confusion on the future with the `result` declared on [line 20](https://github.com/aws-amplify/amplify-ui/pull/2969/files#diff-20eb679d6cd90516423c4c8bec245d4b9d21666124c29bd07a2ecbcd7c1feb93R20), and to make it clear that we are referencing the previous value of `result`, suggest `prevResult`

Reply:
Makes sense,

```suggestion
    setResult((prevResult) => ({ ...prevResult, isLoading: true }));
```